packages/api/app/Models/Permission.js

```diff
@@ -73,9 +74,9 @@ class Permission extends Model {
 				matchedPermission,
 			)
 		) {
-			const technology = await Technology.findOrFail(techonologyResourceId);
-			const technologyOwner = await technology.getOwner();
-			if (!technologyOwner || technologyOwner.id !== user.id) {
+			const technologyInst = await Technology.findOrFail(techonologyResourceId);
```

Comment:
aqui não seria melhor fazer a query direto (count) ? ao inves de buscar o user, verificar os responsibles, etc.

Reply:
Aqui o que eu faço é usar um método pra saber se o usuário em questão é um "responsável", se eu passar apenas o user em `technologyInst.checkResponsible(user);` ele checa por padrão se ele é o owner, opcionalmente eu posso passar um role como segundo parâmetro para ver se ele tem outro papel, nesse caso eu checo apenas se ele é owner da tecnologia. Eu achei melhor assim, mas se for o caso podemos refatorar isso ai.